src/message/pack_encrypted/mod.rs

```diff
@@ -129,6 +134,72 @@ impl Message {
 
         Ok((msg, metadata))
     }
+
+    fn validate(&self, to: &str, from: Option<&str>, sign_by: Option<&str>) -> Result<bool> {
```

Comment:
If it is private method i suggest call it `_validate`. Also i think this `validate` method must be common for any packing method type like `pack_encrypted`, `pack_signed`.